### ISSUE TITLE
style/single-key-in | False positive

### ISSUE THREAD

**Issue by dpassen** [NoahTheDuke/splint#39]
Found what I believe to be a false positive in the `style/single-key-in` rule:

Given the file, `foo.clj`:
```clj
(let [m {}]
  (-> m
      (assoc-in [:user :existing-ids] [:fake-id])))

```

Running splint:
```sh
bash-3.2$ clojure -Sdeps '{:deps {io.github.noahtheduke/splint {:mvn/version "1.23.1"}}}' -M -m noahtheduke.splint foo.clj
foo.clj:3:7 [style/single-key-in] - Use `assoc` instead of recreating it.
(assoc-in [:user :existing-ids] [:fake-id])
Consider using:
(assoc [:user :existing-ids] :fake-id)

Linting took 28ms, checked 1 files, 1 style warnings
```

yields an incorrect suggestion, as the two code snippets are not equivalent. The rule does not trip outside of the thread-first (`->`) style, but this was taken from production code in a larger chain of operations.

**Comment by NoahTheDuke** [comment]
Thanks for the report!
